### PR TITLE
[1.16.5]Fix PartEntity id shifting

### PR DESCRIPTION
**Pull request by XenonDB**
https://github.com/MinecraftForge/MinecraftForge/issues/8188

### CODE REVIEW

File: patches/minecraft/net/minecraft/client/network/play/ClientPlayNetHandler.java.patch [hunk 1]
```diff
@@ -46,7 +46,7 @@
 +            net.minecraftforge.entity.PartEntity<?>[] aenderdragonpartentity = livingentity.getParts();
  
              for(int i = 0; i < aenderdragonpartentity.length; ++i) {
-                aenderdragonpartentity[i].func_145769_d(i + p_147281_1_.func_149024_d());
+                aenderdragonpartentity[i].func_145769_d((i + 1) + p_147281_1_.func_149024_d());
```
Comment (ChampionAsh5357) [inline patches/minecraft/net/minecraft/client/network/play/ClientPlayNetHandler.java.patch]: Should have comment linking to the mc issue this fixes (E.g. ` //Forge: Fixes MC-123456`).

**Review by ChampionAsh5357 — CHANGES_REQUESTED** [review]
As this issue exists in both 1.16 and 1.17. This should be ported to 1.17 and then once merged handled back in 1.16.

**Comment by XenonDB** [comment]
I should find where this code is in 1.17..

**Comment by ChampionAsh5357** [comment]
Within `EnderDragon` as packet data is now handled directly on the entity.

**Comment by XenonDB** [comment]
> Within `EnderDragon` as packet data is now handled directly on the entity.

Ok, I found that.

**Comment by ChampionAsh5357** [comment]
As Ater pointed out to me, this is a manual edit instead of an actual patch. Please properly generate the patch using the `genPatches` task after you setup your Forge project in your IDE of choice.

**Comment by XenonDB** [comment]
hmm...since I delete my fork relative to this. I will close this PR and make a new one.
Is there any way to make a private patched version for testing?

**Comment by ChampionAsh5357** [comment]
> hmm...since I delete my fork relative to this. I will close this PR and make a new one. Is there any way to make a private patched version for testing?

Just run the test client in the dev environment. As long as it is not the clean one, it should have your changes.

**Pull request closed without merging**